x-pack/plugins/observability_onboarding/server/routes/logs/route.ts

```diff
@@ -40,9 +40,22 @@ const installShipperSetupRoute = createObservabilityOnboardingServerRoute({
     scriptDownloadUrl: string;
     elasticAgentVersion: string;
   }> {
-    const { core, plugins } = resources;
+    const { core, plugins, kibanaVersion } = resources;
     const coreStart = await core.start();
 
+    const fleetPluginStart = await plugins.fleet.start();
+    const agentClient = fleetPluginStart.agentService.asInternalUser;
+
+    // If undefined, we will follow fleet's strategy to select latest available version:
+    // for serverless we will use the latest published version, for statefull we will use
+    // current Kibana version
+    const includeCurrentVersion = kibanaVersion.endsWith('-SNAPSHOT')
+      ? false
+      : undefined;
+
+    const elasticAgentVersion =
+      await agentClient.getLatestAgentAvailableVersion(includeCurrentVersion);
+
```

Comment:
I am finding this a bit tough to read. 
What i understand is, if you pass `undefined`, fleet will automatically handle the logic for serverless and stateful. What happens when you pass `false` for stateful and serverless ?

Reply:
If you pass `false`, irrespective of what flags are set in fleet, we are the ones deciding not to use the kibana current version

Reply:
Do you think extending the comment to explain this will help?

Reply:
Yup adding this in code comment would make it easier if someone is revisiting this

Reply:
Addressed in [372ce96c](https://github.com/elastic/kibana/pull/166920/commits/372ce96c5b5d53f1904dac84bd669ef6c4237599).